hbase-server/src/test/java/org/apache/hadoop/hbase/master/store/TestLocalRegionOnTwoFileSystems.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.hadoop.hbase.testclassification.MasterTests;
+import org.apache.hadoop.hbase.testclassification.MediumTests;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.util.CommonFSUtils;
+import org.apache.hadoop.hbase.util.HFileArchiveUtil;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.hbase.thirdparty.com.google.common.collect.Iterables;
+
+@Category({ MasterTests.class, MediumTests.class })
+public class TestLocalRegionOnTwoFileSystems {
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestLocalRegionOnTwoFileSystems.class);
+
+  private static final HBaseCommonTestingUtility HFILE_UTIL = new HBaseCommonTestingUtility();
+
+  private static final HBaseTestingUtility WAL_UTIL = new HBaseTestingUtility();
+
+  private static ChoreService CHORE_SERVICE;
+
+  private static DirScanPool CLEANER_POOL;
+
+  private static LocalRegion REGION;
+
+  private static byte[] CF = Bytes.toBytes("f");
+
+  private static byte[] QUALIFIER = Bytes.toBytes("q");
+
+  private static String REGION_DIR_NAME = "local";
+
+  private static TableDescriptor TD =
+    TableDescriptorBuilder.newBuilder(TableName.valueOf("test:local"))
+      .setColumnFamily(ColumnFamilyDescriptorBuilder.of(CF)).build();
+
+  private static int COMPACT_MIN = 4;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    WAL_UTIL.startMiniCluster(3);
+    Configuration conf = HFILE_UTIL.getConfiguration();
+    conf.setBoolean(MemStoreLAB.USEMSLAB_KEY, false);
+    CHORE_SERVICE = new ChoreService("TestLocalRegionOnTwoFileSystems");
+    CLEANER_POOL = new DirScanPool(conf);
+    Server server = mock(Server.class);
+    when(server.getConfiguration()).thenReturn(conf);
+    when(server.getServerName())
+      .thenReturn(ServerName.valueOf("localhost", 12345, System.currentTimeMillis()));
+    when(server.getChoreService()).thenReturn(CHORE_SERVICE);
+    Path rootDir = HFILE_UTIL.getDataTestDir();
+    CommonFSUtils.setRootDir(conf, rootDir);
+    Path walRootDir = WAL_UTIL.getDataTestDirOnTestFS();
+    FileSystem walFs = WAL_UTIL.getTestFileSystem();
+    CommonFSUtils.setWALRootDir(conf,
+      walRootDir.makeQualified(walFs.getUri(), walFs.getWorkingDirectory()));
+    LocalRegionParams params = new LocalRegionParams();
+    params.server(server).regionDirName(REGION_DIR_NAME).tableDescriptor(TD)
+      .flushSize(TableDescriptorBuilder.DEFAULT_MEMSTORE_FLUSH_SIZE).flushPerChanges(1_000_000)
+      .flushIntervalMs(TimeUnit.MINUTES.toMillis(15)).compactMin(COMPACT_MIN).maxWals(32)
+      .useHsync(false).ringBufferSlotCount(16).rollPeriodMs(TimeUnit.MINUTES.toMillis(15))
+      .archivedWalSuffix(LocalStore.ARCHIVED_WAL_SUFFIX)
+      .archivedHFileSuffix(LocalStore.ARCHIVED_HFILE_SUFFIX);
+    REGION = LocalRegion.create(params);
```

Comment:
Good

Reply:
Are there params to set for local regions walfs and hfilefs?

Reply:
No, there are set in Configuration.

Reply:
Where are the keys defined?